pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -737,11 +732,11 @@ public static void removeColumnMetadataInfo(PropertiesConfiguration properties,
    * @return raw index creator
    * @throws IOException
    */
-  public static ForwardIndexCreator getRawIndexCreatorForColumn(File file,
-      ChunkCompressionType compressionType, String column, DataType dataType, int totalDocs,
-      int lengthOfLongestEntry, boolean deriveNumDocsPerChunk, int writerVersion)
+  public static ForwardIndexCreator getRawIndexCreatorForColumn(File file, ChunkCompressionType compressionType,
+      String column, DataType dataType, int totalDocs, int lengthOfLongestEntry, boolean deriveNumDocsPerChunk,
+      int writerVersion)
       throws IOException {
-    switch (dataType) {
+    switch (dataType.getStoredType()) {
```

Comment:
Can you please add some comments in DataType.java explaining the usage DataType vs stored DataType, i.e when should DataType be used and when should DataType.getStoredType() be used?

Reply:
Added some comments in the javadoc of `DataType.getStoredType()`